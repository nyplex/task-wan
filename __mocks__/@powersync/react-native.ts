module.exports = {
  PowerSyncDatabase: jest.fn().mockImplementation(() => ({
    connect: jest.fn(),
    ready: true,
  })),
};
