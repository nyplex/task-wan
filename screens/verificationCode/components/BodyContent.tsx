import Animated, {
  useAnimatedStyle,
  interpolate,
} from "react-native-reanimated";
import { useReanimatedKeyboardAnimation } from "react-native-keyboard-controller";
import { Box } from "@/components/gluestack/box";
import Text from "@/components/Text";

const BodyContent = () => {
  const { progress } = useReanimatedKeyboardAnimation();

  const animatedStyles = useAnimatedStyle(() => {
    const scale = interpolate(progress.value, [0, 1], [1, 0.01]);
    const height = interpolate(progress.value, [0, 1], [200, 10]);
    const opacity = interpolate(progress.value, [0, 1], [1, 0.5]);

    return {
      height,
      opacity,
      transform: [{ scale }],
    };
  });

  return (
    <Box>
      <Text className="text-center mt-12" size="body" weight="semi-bold">
        Verify Account
      </Text>

      <Animated.View
        style={[
          animatedStyles,
          { height: 200, width: 250, alignSelf: "center", marginTop: 40 },
        ]}
      >
        <Animated.Image
          source={require("@/assets/images/verifyEmail.png")}
          style={[{ width: 250, height: "100%" }, animatedStyles]}
        />
      </Animated.View>

      <Text className="text-center mt-8 mb-4" size="bodyS" weight="regular">
        Please enter the verification number we sent to your email
      </Text>
    </Box>
  );
};

export default BodyContent;
