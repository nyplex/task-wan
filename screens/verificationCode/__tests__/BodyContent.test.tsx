import React from "react";
import { render } from "@testing-library/react-native";
import BodyContent from "../components/BodyContent";

// Mock Image import
jest.mock("@/assets/images/verifyEmail.png", () => "verifyEmail.png");

// Mock Animated and interpolation logic
jest.mock("react-native-reanimated", () => require("react-native-reanimated/mock"));

// Mock keyboard animation hook
jest.mock("react-native-keyboard-controller", () => ({
  useReanimatedKeyboardAnimation: () => ({
    progress: { value: 0 },
  }),
}));

describe("BodyContent", () => {
  it("renders correctly and matches snapshot", () => {
    const { toJSON, getByText } = render(<BodyContent />);

    // Check essential content
    expect(getByText("Verify Account")).toBeTruthy();
    expect(getByText("Please enter the verification number we sent to your email")).toBeTruthy();

    // Snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
