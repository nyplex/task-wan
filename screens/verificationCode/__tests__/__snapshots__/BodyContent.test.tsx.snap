// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyContent renders correctly and matches snapshot 1`] = `
<View>
  <Text
    className="font-body based-styles font-semibold text-typography-primary text-[16px] tracking-[0px] text-center mt-12"
  >
    Verify Account
  </Text>
  <View
    style={
      [
        {
          "height": undefined,
          "opacity": undefined,
          "transform": [
            {
              "scale": undefined,
            },
          ],
        },
        {
          "alignSelf": "center",
          "height": 200,
          "marginTop": 40,
          "width": 250,
        },
      ]
    }
  >
    <Image
      source="verifyEmail.png"
      style={
        [
          {
            "height": "100%",
            "width": 250,
          },
          {
            "height": undefined,
            "opacity": undefined,
            "transform": [
              {
                "scale": undefined,
              },
            ],
          },
        ]
      }
    />
  </View>
  <Text
    className="based-styles font-regular text-typography-primary text-[14px] tracking-[0px] text-center mt-8 mb-4"
  >
    Please enter the verification number we sent to your email
  </Text>
</View>
`;
